xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -657,15 +712,30 @@ void CGUIDialogPVRTimerSettings::AddCondition(
   setting->SetDependencies(deps);
 }
 
-int CGUIDialogPVRTimerSettings::GetDateAsInt(const CDateTime &datetime)
+int CGUIDialogPVRTimerSettings::GetDateAsIndex(const CDateTime &datetime)
 {
-  const CDateTime date(datetime.GetYear(), datetime.GetMonth(), datetime.GetDay(), 0, 0, 0);
   time_t t(0);
-  date.GetAsTime(t);
+  const CDateTime date(datetime.GetYear(), datetime.GetMonth(), datetime.GetDay(), 0, 0, 0);
+  if (date > m_maxDateCovered)
+    (m_maxDateCovered + CDateTimeSpan(1, 0, 0, 0)).GetAsTime(t);
+  else if (date < m_minDateCovered)
+    (m_minDateCovered - CDateTimeSpan(1, 0, 0, 0)).GetAsTime(t);
+  else
+    date.GetAsTime(t);
```

Comment:
This allows for the 'no change' < and > list options - which coupled with the 'else' cases above is the clever bit.
It allows us to tolerate (and edit) timers with out of range dates without affecting these dates unless the user specifically selects a value in the spinner's valid date range.